.github/workflows/UnitTesting.yaml

```
@@ -9,29 +9,30 @@ on:
 
 jobs:
   test:
-    # "setup-python" action doesn't provide python 3.4 binaries for ubuntu-latest.
-    # Sticking to Ubuntu 18.04 as recommended here:
-    # https://github.com/actions/setup-python/issues/185#issuecomment-768232756
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-22.04
     env:
-      py27: 2.7
-      py34: 3.4
-      py35: 3.5
-      py36: 3.6
       py37: 3.7
       py38: 3.8
       py39: 3.9
       py310: '3.10'
       py311: '3.11'
+      DB_DATABASE: test_db
+      DB_USER: root
+      DB_PASSWORD: root
     strategy:
       fail-fast: false
       matrix:
-        python-version: [py27, py34, py35, py36, py37, py38, py39, py310, py311]
+        python-version: [py37, py38, py39, py310, py311]
         testenv: [core, ext]
     steps:
       - name: Checkout repo
         uses: actions/checkout@v3
 
+      - name: Start MySQL
+        run: |
```

add 
```
if: ${{ matrix.testenv == 'ext' }}
```
between name and run

---

Protip @wangzlei , you can use suggestions for this, e.g.

```suggestion
        if: ${{ matrix.testenv == 'ext' }}
        run: |
```